OpenMTRDemo/Forms/ExpandedImageDialog.cs

```diff
@@ -3,46 +3,67 @@
 using OpenCvSharp;
 using OpenMTRDemo.Forms;
 
-namespace OpenMTRDemo
+namespace OpenMTRDemo.Forms
 {
-    public partial class MainForm : BaseForm
+    public partial class ExpandedImageDialog : BaseForm
     {
-        OpenFileDialog OpenBrowser;
-        SaveFileDialog SaveBrowser;
-        OpenMTR.Meter meter;
+        public Mat Source, Image;
 
-        public MainForm()
+        public ExpandedImageDialog(Mat source, Mat image)
         {
-            OpenBrowser = new OpenFileDialog();
-            SaveBrowser = new SaveFileDialog();
-            OpenBrowser.Filter = "Image Files|*.bmp;*.gif;*.jpg;*.jpeg;*.png";
-            SaveBrowser.Filter = "JPG|*.jpg";
+            this.Source = source;
+            this.Image = image;
             InitializeComponent();
+            this.DialogResult = DialogResult.Cancel;
+            InputImageBox.Image = DemoUtilities.MatToBitmap(Source);
+            OutputImageBox.Image = DemoUtilities.MatToBitmap(Image);
+        }
+
+        public ExpandedImageDialog(Mat source, Mat image, string[] filterList)
+        {
+            this.Source = source;
+            this.Image = image;
+            InitializeComponent();
+            this.DialogResult = DialogResult.Cancel;
+            foreach (string filter in filterList)
+            {
+                if (FilterListBox.Items.Contains(filter))
+                {
+                    FilterListBox.SelectedItems.Add(filter);
+                }
+            }
+            InputImageBox.Image = DemoUtilities.MatToBitmap(Source);
+            Render();
+        }
+
+        public override void saveToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            SaveFileDialog save = new SaveFileDialog();
```

Comment:
Create a class level instance of loadSaveDialog for this form to use.

Reply:
Oh, right... I totally forget we had that.